mypy/checkexpr.py

```diff
@@ -1806,123 +1808,278 @@ def get_operator_method(self, op: str) -> str:
                                             self.named_type, self.not_ready_callback, local_errors,
                                             original_type=base_type, chk=self.chk)
+        return self.check_op_local(method, method_type, base_type, arg, context, local_errors)
+
+    def check_op_local(self,
+                       method_name,
+                       method_type: Type,
+                       base_type: Type,
+                       arg: Expression,
+                       context: Context,
+                       local_errors: MessageBuilder) -> Tuple[Type, Type]:
+        """Type check a binary operation using the (assumed) type of the operator method.
+
+        Return tuple (result type, inferred operator method type).
+        """
         callable_name = None
         object_type = None
         if isinstance(base_type, Instance):
             # TODO: Find out in which class the method was defined originally?
             # TODO: Support non-Instance types.
-            callable_name = '{}.{}'.format(base_type.type.fullname(), method)
+            callable_name = '{}.{}'.format(base_type.type.fullname(), method_name)
             object_type = base_type
         return self.check_call(method_type, [arg], [nodes.ARG_POS],
                                context, arg_messages=local_errors,
                                callable_name=callable_name, object_type=object_type)
 
+    def check_op_reversible(self,
+                            op_name: str,
+                            left_type: Type,
+                            left_expr: Expression,
+                            right_type: Type,
+                            right_expr: Expression,
+                            context: Context) -> Tuple[Type, Type]:
+        # Note: this kludge exists mostly to maintain compatibility with
+        # existing error messages. Apparently, if the left-hand-side is a
+        # union and we have a type mismatch, we print out a special,
+        # abbreviated error message. (See messages.unsupported_operand_types).
+        unions_present = isinstance(left_type, UnionType)
+
+        def make_local_errors() -> MessageBuilder:
+            """Creates a new MessageBuilder object."""
+            local_errors = self.msg.clean_copy()
+            local_errors.disable_count = 0
+            if unions_present:
+                local_errors.disable_type_names += 1
+            return local_errors
+
+        def lookup_operator(op_name: str, base_type: Type) -> Optional[Type]:
+            """Looks up the given operator and returns the corresponding type,
+            if it exists."""
+            local_errors = make_local_errors()
+
+            # TODO: Remove this call and rely just on analyze_member_access
+            # Currently, it seems we still need this to correctly deal with
+            # things like metaclasses?
+            #
+            # E.g. see the pythoneval.testMetaclassOpAccessAny test case.
```

Comment:
Could you please add details about this to https://github.com/python/mypy/issues/5136?